commons/src/main/java/io/aiven/kafka/connect/common/config/AivenCommonConfig.java

```diff
@@ -68,8 +139,17 @@ private void validate() {
                     FORMAT_OUTPUT_ENVELOPE_CONFIG, false, FORMAT_OUTPUT_FIELDS_CONFIG);
             throw new ConfigException(msg);
         }
+        if (getCustomRecordGrouperBuilder() == null) {
+            // if there is a custom record grouper builder, it will validate the filename template
+            new FilenameTemplateValidator(FILE_NAME_TEMPLATE_CONFIG).ensureValid(FILE_NAME_TEMPLATE_CONFIG,
+                    getString(FILE_NAME_TEMPLATE_CONFIG));
+        }
         validateKeyFilenameTemplate();
     }
+    protected static void addExtensionConfig(final ConfigDef configDef) {
+        final ServiceLoader<ExtraConfiguration> extraConfigurations = ServiceLoader.load(ExtraConfiguration.class);
+        extraConfigurations.forEach(extraConfiguration -> extraConfiguration.configure(configDef));
```

Comment:
Is being present on the classpath a sufficient condition for inclusion?
When used in test environments with test ExtraConfigurations, you wouldn't be able to turn them off.
If someone packaged multiple plugins together, each with their own specific ExtraConfigurations, how would each connector get the configurations specific to them?

Reply:
I would expect that a suer would package only the ones that were needed

You could add a a parameter to the request that specifies the additional groups, but that would mean that those extra config require the user to request them, so maybe `Configurable` would specify which additional configuration extras eare needed

Not sure if that would work though. WDYT